src/ex_getln.c

```diff
@@ -4250,7 +4250,7 @@ get_prompt(void)
     void
 set_prompt(char_u* str)
 {
-    vim_strncpy(current_prompt, str, sizeof(current_prompt) - 1);
+    current_prompt = str;
```

Comment:
Thank you for the improvements.

I think this code is better.

```suggestion
    VIM_CLEAR(current_prompt);
    current_prompt = vim_strsave(str);
```